src/sinks/splunk_hec/common/acknowledgements.rs

```diff
@@ -26,6 +27,13 @@ pub struct HecClientAcknowledgementsConfig {
     pub query_interval: NonZeroU8,
     pub retry_limit: NonZeroU8,
     pub max_pending_acks: NonZeroU64,
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        flatten,
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub inner: AcknowledgementsConfig,
```

Comment:
With this added, I think we can remove/deprecate the `indexer_acknowledgements_enabled` option (replacing its use with `inner.enabled`).

Reply:
Oh, I thought we would want to separate the indexer acknowledgements from the end-to-end acknowledgements. I'll raise that as a follow-up issue once this is merged.

Reply:
Mm, perhaps I have the wrong idea here. Currently if `indexer_acknowledgements_enabled` is `true` we integrate acknowledge based on indexer acks and if `false` we acknowledge based solely on the `200` response. What will the effect of `inner.enabled` `true/false` be?

Reply:
The `AcknowledgementConfig` is not used in the sink itself. Rather it is used, when enabled, to turn on acknowledgements on all connected sources.

Reply:
Aha, right -- had things mixed up conceptually. That makes sense, never mind what I said then.